internal/factsengine/gatherers/sysctl.go

```diff
@@ -0,0 +1,118 @@
+package gatherers
+
+import (
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+	"github.com/trento-project/agent/pkg/utils"
+)
+
+const (
+	SysctlGathererName = "sysctl"
+)
+
+// nolint:gochecknoglobals
+var (
+	SysctlValueNotFound = entities.FactGatheringError{
+		Type:    "sysctl-value-not-found",
+		Message: "requested value not found in sysctl output",
+	}
+
+	SysctlCommandError = entities.FactGatheringError{
+		Type:    "sysctl-cmd-error",
+		Message: "error executing sysctl command",
+	}
+
+	SysctlMissingArgument = entities.FactGatheringError{
+		Type:    "sysctl-missing-argument",
+		Message: "missing required argument",
+	}
+)
+
+type SysctlGatherer struct {
+	executor utils.CommandExecutor
+}
+
+func NewDefaultSysctlGatherer() *SysctlGatherer {
+	return NewSysctlGatherer(utils.Executor{})
+}
+
+func NewSysctlGatherer(executor utils.CommandExecutor) *SysctlGatherer {
+	return &SysctlGatherer{
+		executor: executor,
+	}
+}
+
+func (s *SysctlGatherer) Gather(factsRequests []entities.FactRequest) ([]entities.Fact, error) {
+	facts := []entities.Fact{}
+	log.Infof("Starting sysctl facts gathering process")
```

Comment:
We could use the string interpolation with `SysctlGathererName` as we do with many others.
Small change, but it would help us if we for some reason change the gatherer name, to not forget this details